### ISSUE TITLE
How to customize a hash serialization?

### ISSUE THREAD

**Issue by jlecour** [rails-api/active_model_serializers#181]
Hi,

I can't find a good way to do a custom serialization of a plain old Hash.

Here is an example : 

``` ruby
class MyObject
  include ActiveModel::SerializerSupport

  attr_reader :name, :age, :address, :settings

  def initialize(name, age, address, settings = {})
    @name = name
    @age = age
    @address = address
    @settings = settings.merge(:timezone => 'GMT', :other => 'xyz')
  end

end

class MyObjectSerializer < ActiveModel::Serializer
  attributes :name, :age, :settings
end

MyObjectSerializer.new(MyObject.new('toto', 12, 'mars', {:timezone => 'Paris', :foo => 'bar'})).as_json

# => {:my_object=>{:name=>"toto", :age=>12, :settings=>{:timezone=>"GMT", :foo=>"bar", :other=>"xyz"}}}
```

I'd like not to serialize all the key/values of the internal settings Hash, so I make a custom Serializer

``` ruby
class MyObjectSerializer < ActiveModel::Serializer
  attributes :name, :age, :settings

  def settings
    SettingsSerializer.new(object.settings).as_json
  end
end

class SettingsSerializer < ActiveModel::Serializer
  attributes :timezone
end
```

But I get an error : `NoMethodError: undefined method`read_attribute_for_serialization' for {:timezone=>"GMT", :foo=>"bar", :other=>"xyz"}:Hash`

So I have a few bad choices (as far as I can tell) :
1. I can add a custom `SettingsSerializer#attributes` method with only the keys I want, but the natural DSL of AMS (using `attributes` in the class definition) is useless
2. I can make a value object to embed the hash and use a regular serializer but it seems a bit overkill
3. I can just use a hash which I delete the undesired keys from (for example with `Hash#only(*keys)` but I don't take advantage of the power of AMS.

Is there a proper/clean way to do this or not yet?

Thanks

**Comment by therealjessesanford** [comment]
In case anyone finds this via google I have written a SO answer over here that explains how to use PORO hashes with AMS: http://stackoverflow.com/a/34056668/511168

**Comment by bf4** [comment]
@therealjessesanford This is fixed in 0.10. I'll reply in SO.

**Comment by icem** [comment]
I have found an easy way to make serializer for hash:
```ruby
class LocationSerializer < ActiveModel::Serializer
  attributes :lat, :lon, :city
  attribute :address do
    # Example some custom key 
  end

  def read_attribute_for_serialization(attr)
    object[attr.to_s]
  end
end

```
My hash example `{'lat' => 53.3, 'lon' => 0.19, 'city' => 'London', ...}`
I can even use my LocationSerializer through `has_one` on regular serializer, works good.
[for v10.6]
